sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -190,6 +190,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val CONSTRAINT_PROPAGATION_ENABLED = buildConf("spark.sql.constraintPropagation.enabled")
+    .internal()
```

Comment:
Due to the fact there are few users reporting hitting this issue, we may need to expose it as an external flag.

However, I would think that a large portion of external users may not know constraint propagation. It might not be intuitive to link the problem they hit to constraint propagation and to find this config, even it is external.